utils/fs/directory.cpp

```diff
@@ -198,16 +192,12 @@ struct utils::fs::detail::directory_iterator::impl : utils::noncopyable {
     {
         ::dirent* result;
 
-        if (::readdir_r(_dirp, &_dirent, &result) == -1) {
-            const int original_errno = errno;
-            throw fs::system_error(F("readdir_r(%s) failed") % _path,
-                                   original_errno);
-        }
+        result = ::readdir(_dirp);
```

Comment:
Merge initialization with definition, which in turn lets you declare the variable as const.